src/main/java/executor/service/service/StepExecutionSleep.java

```diff
@@ -0,0 +1,29 @@
+package executor.service.service;
+
+import executor.service.model.StepDTO;
+import org.openqa.selenium.WebDriver;
+
+public class StepExecutionSleep implements StepExecution{
+
+    private static final String ACTION = "sleep";
+    @Override
+    public String getStepAction() {
+        return ACTION;
+    }
+
+    @Override
+    public void step(WebDriver webDriver, StepDTO stepDTO) throws InterruptedException, IllegalArgumentException {
+        try {
+            int sleepTime = Integer.parseInt(stepDTO.getValue());
+            Thread.sleep(sleepTime);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("Sleep argument must be an integer");
+        } catch (InterruptedException e) {
+            throw e;
```

Comment:
And in this case, our `StepExecutionSleep` service overrides `void step(..)` method that doesn't throw any checked exceptions.
In such case we can preserve interruption by restoring the interrupted status:
`Thread.currentThread().interrupt();` instead of throwing `InterruptedException`
```suggestion
    public void step(WebDriver webDriver, StepDTO stepDTO) {
        try {
            int sleepTime = Integer.parseInt(stepDTO.getValue());
            Thread.sleep(sleepTime);
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException("Sleep argument must be an integer");
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
```